src/index.ts

```diff
@@ -1,86 +1,94 @@
   const https = require("https");
   const path = require("path");
   const config = require("./infrastructure/config");
-  const helmet = require("helmet");
-  const sanitization = require("login.dfe.sanitization");
   const healthCheck = require("login.dfe.healthcheck");
   const { getErrorHandler } = require("login.dfe.express-error-handling");
-  const routes = require("./routes");
 
-  try {
-    const dbConnection = await createConnection();
+  var dbMigrationPromise = connection.then(c =>
     // Run migrations on start up; only boot up if migrations succeed!
-    dbConnection.runMigrations({ transaction: true });
-  } catch (err) {
-    throw new Error(`Could not boot application, failed to connect to database, error: ${err}`);
-  }
-
-  const app = express();
-  app.use(
-    helmet({
-      noCache: true,
-      frameguard: {
-        action: "deny"
-      }
-    })
+    c.runMigrations({ transaction: true })
   );
 
-  const appViews = path.join(__dirname, "views");
-
-  nunjucks.configure(appViews, {
-    autoescape: true,
-    express: app,
-    noCache: true,
-    watch: true
+  dbMigrationPromise.catch(err => {
+    throw new Error("Could not migrate database");
   });
 
-  app.use(bodyParser.urlencoded({ extended: true }));
-  app.use(sanitization());
-  app.set("view engine", "html");
+  dbMigrationPromise.then(() => {
+    var app = express();
+    const appViews = path.join(__dirname, "views");
 
-  app.use(
-    "/healthcheck",
-    healthCheck({
-      config
-    })
-  );
+    nunjucks.configure(appViews, {
+      autoescape: true,
+      express: app,
+      noCache: true,
+      watch: true
+    });
 
-  // Error handing
-  app.use(
-    getErrorHandler({
-      logger
-    })
-  );
+    app.use(bodyParser.urlencoded({ extended: true }));
+    app.set("view engine", "html");
 
-  app.use("/", routes);
+    app.use(
+      helmet({
+        noCache: true,
+        frameguard: {
+          action: "deny"
+        }
+      })
+    );
 
-  if (config.hostingEnvironment.env === "dev") {
-    app.proxy = true;
+    app.use(
+      "/healthcheck",
+      healthCheck({
+        config
+      })
+    );
 
-    const options = {
-      key: config.hostingEnvironment.sslKey,
-      cert: config.hostingEnvironment.sslCert,
-      requestCert: false,
-      rejectUnauthorized: false
-    };
-    const server = https.createServer(options, app);
+    // Error handing
+    app.use(
+      getErrorHandler({
+        logger
+      })
+    );
 
-    server.listen(config.hostingEnvironment.port, () => {
-      logger.info(
-        `Dev server listening on https://${config.hostingEnvironment.host}:${config.hostingEnvironment.port}`
-      );
-    });
-  } else {
-    app.set("trust proxy", 1);
-    app.listen(process.env.PORT, () => {
-      logger.info(`Server listening on http://${config.hostingEnvironment.host}:${config.hostingEnvironment.port}`);
+    useExpressServer(app, {
+      controllers: [__dirname + "/controllers/*.js"]
```

Comment:
This seems magic-y...

Reply:
which part, the file scan?

Reply:
Yeah! It's weird that the controllers are not being explicitly imported. I mean, we do the same thing for the views, but those are HTML files so it makes sense they're not `require`d.

I quite like that in JS projects generally stuff is explicitly imported, nothing enters the global namespace by accident, etc. Even if it's more verbose / involves writing a bit of boilerplate.

Reply:
I guess it's subject to taste - I opted for the scan because this removes the need to remember an additional place to change when introducing a new controller, following the conventions I know from ASP.NET. I'd quite like to keep it the way it is but am happy to be overruled. @dankmitchell can you cast a vote?

Reply:
Yeah, the need to add an additional line to include a controller is something I personally am fond of, but am happy to leave it up to vote. Dan to break the tie :)

Reply:
Ship it